src/hooks/useRole.ts

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { useK8sWatchResource } from '../k8s';
+import { ConfigMapGroupVersionKind, ConfigMapModel } from '../models';
+import { ConfigMap } from '../types/configmap';
+
+export const useRoleMap = (watch?: boolean): [object, boolean, unknown] => {
+  const {
+    data: configMap,
+    isLoading: roleLoading,
+    error,
+  } = useK8sWatchResource<ConfigMap>(
+    {
+      groupVersionKind: ConfigMapGroupVersionKind,
+      namespace: 'konflux-info',
+      name: 'konflux-public-info',
+      watch,
+    },
+    ConfigMapModel,
+  );
```

Comment:
create a separate hook for fetching the config map and parsing the data from the config map. Also, Create a type for the parsed data from configmap.